rosbag2_cpp/include/rosbag2_cpp/converter_interfaces/serialization_format_deserializer.hpp

```diff
@@ -37,6 +39,21 @@ class SerializationFormatDeserializer
     std::shared_ptr<const rosbag2_storage::SerializedBagMessage> serialized_message,
     const rosidl_message_type_support_t * type_support,
     std::shared_ptr<rosbag2_introspection_message_t> ros_message) = 0;
+
+  /**
+   * Get the deserializer package name
+   */
+  static std::string get_package_name()
+  {
+    return "rosbag2_cpp";
+  }
+  /**
+   * Get the deserializer base class name
+   */
+  static std::string get_base_class_name()
+  {
+    return ConverterTraits<SerializationFormatDeserializer>::name;
```

Comment:
This traits thing looks redundant if you're building this value into the class - the reason it was needed was so that it could be "indexed" on the class type. Now that the value is attached directly to the class, no reason to have the separate index. You can use `get_base_class_name` in the factory impl as well instead of `Traits`.

Unless I'm missing something, I think my request is - remove `ConverterTraits` entirely. The same thing should be possible for `CompressionTraits`

Reply:
My original thought when adding the `*Traits` headers was that it would help standardize around the implementation used by `StorageTraits`. But now with the static `get_base_class_name()` function, it definitely makes sense to get rid of all the `*Traits` files.